### PR TITLE
Groups analytics support

### PR DESCRIPTION
**Pull request by jsuarezb**
Add group analytics support within the `Mixpanel::Groups` class.

### CODE REVIEW

File: lib/mixpanel-ruby/groups.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        '$group_key' => group_key,
+        '$group_id' => group_id,
+        '$unset' => properties,
+      }.merge(optional_params)
+      message['$ip'] = ip if ip
+
+      update(message)
+    end
+
+    # Permanently delete a group from \Mixpanel groups analytics
```
Comment (imcleary) [inline lib/mixpanel-ruby/groups.rb]: nit: deletes the group profile (all group properties on events stay)

File: lib/mixpanel-ruby/groups.rb [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      rescue MixpanelError => e
+        @error_handler.handle(e)
+        ret = false
+      end
+
+      ret
+    end
+
+    private
+
```
Comment (imcleary) [inline lib/mixpanel-ruby/groups.rb]: you're missing one update function: $remove, which removes a value from a list-valued property (if present). Spec is `groupUpdates.remove(property name, value)`.

Comment (jsuarezb) [reply lib/mixpanel-ruby/groups.rb]: True! There it is. :wink:

**Comment by jsuarezb** [comment]
I'm not seeing any support for superproperties within the gem. It would be a nice feature though, without a doubt.

**Comment by tdumitrescu** [comment]
The problem with features like superproperties/identify in Mixpanel's server-side SDKs is that they are typically used in a multi-tenant way. E.g., in a Rails app a single `Mixpanel::Tracker` instance will be shared across different requests with different users. Since MP superproperties are purely a client-side concept, if you wanted to preserve the same properties for the same user across different requests, you'd need to provide your own backing data store for those associations. And if you have multiple appserver instances, that will also need to be a shared store like a SQL DB or Redis. At that point we're pretty far beyond the scope of what to include in a generic tracking library...

**Comment by imcleary** [comment]
Oof, noted. In that case, lgtm.

**Comment by andrewpurcell** [comment]
Thanks for your contribution @jsuarezb!

**Comment by jsuarezb** [comment]
A pleasure :grin:

**Comment by andrewpurcell** [comment]
https://rubygems.org/gems/mixpanel-ruby/versions/2.2.2

**Pull request merged**